parser.ts

```diff
@@ -398,9 +456,32 @@ export function traverseStmt(c: TreeCursor, s: string): Stmt<null> {
   }
 }
 
+export function traverseBracketType(c: TreeCursor, s: string): Type {
+  // For now, always a VariableName
+  let bracketTypes = [];
+  c.firstChild();
+  while (c.nextSibling()) {
+    bracketTypes.push(traverseType(c, s));
+    c.nextSibling();
+  }
+  c.parent();
+  if (bracketTypes.length == 1) {
+    //List
+    return LIST(bracketTypes[0]);
+  } else if (bracketTypes.length == 2) {
+    //Dict?
+  } else {
+    throw new Error(
+      "Can Not Parse Type " + s.substring(c.from, c.to) + " " + c.node.from + " " + c.node.to
+    );
+  }
+}
+
 export function traverseType(c: TreeCursor, s: string): Type {
   // For now, always a VariableName
+
   let name = s.substring(c.from, c.to);
+  if (name.includes("[")) return traverseBracketType(c, s);
```

Comment:
This seems brittle. Is there a specific tag name or test we can do in order to detect this? I see that `traverseBracketType` goes into children, so presumably it has a recognizable name other than substring-ing?

Reply:
We're changing this now, we can use `ArrayExpression` to recognize things like `[int]` as the type.